lib/mongoid/orderable/helpers.rb

```diff
@@ -37,11 +33,18 @@ def orderable_scope_changed_query(column)
 
       def bottom_orderable_position(column = nil)
         column ||= default_orderable_column
-        @bottom_orderable_position = begin
-          positions_list = orderable_scoped(column).distinct(orderable_column(column)).compact
-          return orderable_base(column) if positions_list.empty?
-          max = positions_list.map(&:to_i).max.to_i
-          in_list?(column) ? max : max.next
+        col = orderable_column(column)
+        max = orderable_scoped(column).ne(col => nil).max(col)
+        return orderable_base(column) unless max
+        in_list?(column) ? max : max.next
+      end
+
+      # Prevents usage of identity map in Mongoid 3
```

Comment:
You could use a mongoid-compatibility `if` here and avoid having to evaluate the `if` at runtime.